evaluation/evaluator.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from pathlib import Path
+import torch
+
+from models.utils import (compute_pose_error, compute_epipolar_error,
+                          estimate_pose, AverageTimer, pose_auc,
+                          rotate_intrinsics, rotate_pose_inplane,
+                          scale_intrinsics)
+from models.matching import Matching
+
+
+class Evaluator:
+    def __init__(self, models_config):
+        self.device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        self.config = models_config
+        self.pose_errors = []
+        self.precisions = []
+        self.matching_scores = []
+
+    def evaluate(self, dataset):
+        print('Running inference on device \"{}\"'.format(self.device))
+        timer = AverageTimer(newline=True)
+
+        matching = Matching(self.config).eval().to(self.device)
+
+        for i, data_dict in enumerate(dataset):
+            # Load the image pair
+            img1, img2 = data_dict['image0'].unsqueeze(0).to(self.device), data_dict['image1'].unsqueeze(0).to(self.device)
+            transf_type, H = data_dict['transformation']['type'], data_dict['transformation']['H']
```

Comment:
Remove this line. We will not have homography transformation at evaluation time.